src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -354,7 +353,11 @@ public Task InitializeAsync(DatabaseRecord record)
                 _initialized = true;
             }
 
-            return Task.Factory.StartNew(HandleDatabaseRecordChange, TaskCreationOptions.LongRunning);
+            return Task.Factory.StartNew(() =>
+            {
+                OpenIndexes(record);
+                HandleDatabaseRecordChange();
+            }, TaskCreationOptions.LongRunning);
```

Comment:
Why is this long running? That will force the thread pool to start a new thread.